moler/device/junipergeneric.py

```diff
@@ -0,0 +1,374 @@
+                JuniperGeneric.proxy_pc: JuniperGeneric.unix_local,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local,
+            },
+            JuniperGeneric.cli: {
+                JuniperGeneric.not_connected: JuniperGeneric.proxy_pc,
+                JuniperGeneric.unix_local: JuniperGeneric.proxy_pc,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local,
+            },
+            JuniperGeneric.configure: {
+                JuniperGeneric.unix_local: JuniperGeneric.cli,
+                JuniperGeneric.proxy_pc: JuniperGeneric.cli,
+                JuniperGeneric.not_connected: JuniperGeneric.cli,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local
+            },
+            JuniperGeneric.unix_local: {
+                JuniperGeneric.cli: JuniperGeneric.proxy_pc,
+                JuniperGeneric.configure: JuniperGeneric.proxy_pc,
+            },
+            JuniperGeneric.proxy_pc: {
+                JuniperGeneric.configure: JuniperGeneric.cli,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local
+            }
+        }
+        return state_hops
+
+    def _prepare_state_hops_without_proxy_pc(self):
+        state_hops = {
+            UnixLocal.not_connected: {
+                JuniperGeneric.cli: UnixLocal.unix_local,
+                JuniperGeneric.configure: UnixLocal.unix_local,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local,
+
+            },
+            UnixLocal.unix_local: {
+                JuniperGeneric.configure: JuniperGeneric.cli
+            },
+            JuniperGeneric.cli: {
+                UnixLocal.not_connected: UnixLocal.unix_local,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local,
+            },
+            JuniperGeneric.configure: {
+                UnixLocal.unix_local: JuniperGeneric.cli,
+                UnixLocal.not_connected: JuniperGeneric.cli,
+                JuniperGeneric.unix_local_root: JuniperGeneric.unix_local,
+
+            },
+        }
+        return state_hops
+
+    def _get_packages_for_state(self, state, observer):
+        available = {UnixLocal.cmds: [], UnixLocal.events: []}
+        if state == UnixLocal.unix_local or state == JuniperGeneric.proxy_pc:
+            available = {UnixLocal.cmds: ['moler.cmd.unix'],
+                         UnixLocal.events: ['moler.events.unix']}
+        elif state == JuniperGeneric.cli:
+            available = {UnixLocal.cmds: ['moler.cmd.unix', 'moler.cmd.juniper.cli'],
+                         UnixLocal.events: ['moler.events.unix', 'moler.events.juniper']}
+        elif state == JuniperGeneric.configure:
+            available = {UnixLocal.cmds: ['moler.cmd.juniper.configure'],
+                         UnixLocal.events: ['moler.events.juniper']}
+        return available[observer]
+
+    def _execute_command_to_change_state(self, source_state, dest_state, timeout=-1):
```

Comment:
what's the difference to the method _execute_command_to_change_state in UnixLocal?